packages/compass-logging/src/index.ts

```diff
@@ -58,7 +61,22 @@ export function createLoggerAndTelemetry(component: string): {
     event: string,
     properties: TrackProps = {}
   ): Promise<void> => {
-    const { trackUsageStatistics } = await preferencesIpc.getPreferences();
+    // Avoid circular dependency between compass-logging and compass-preferences-model
+    // Note that this is mainly a performance optimization, since the main process
+    // telemetry code also checks this preference value, so it is safe to fall back to 'true'.
+    try {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore-error Types from the dependency may not be available in early bootstrap.
+      preferencesIpc ??= (await import('compass-preferences-model'))
+        .preferencesIpc;
```

Comment:
Small q for my understanding: as compass-logging is something we should be able to use both in main process and renderer process, shouldn't this somehow fallback to reading settings from actual model? Seems like `preferencesIpc` will just always return empty preferences object when used outside of renderer, or the idea here that tracking in main should just always be disabled? 🤔

Reply:
Ugh yes, good point. For now, I’ve simply defaulted this to `true`, but it *is* really annoying that there is no unified access to the preferences model that works both in the renderer and the main process, in the way that e.g. compass-logging works well in both. Don’t want to bloat this PR further since it’s not entirely trivial to get access to the main preferences instance, but I wrote down some notes in COMPASS-6149 as a TODO ticket here.

Reply:
Thanks for context, good to know we will be addressing this later!

One thing I still don't really follow is how this defaults to `true` (sorry, I am probably missing something here 🙇). AFAIU this import will not fail in node or electron main environment and the `preferencesIpc` fallback currently will return an empty object:

![image](https://user-images.githubusercontent.com/5036933/191061160-0ab517a5-208f-4ef2-927c-092bbc417c08.png)

so all places where we check for `trackUsageStatistics` will evaluate to `false`, right?

Reply:
It explicitly defaults to `true` here: https://github.com/mongodb-js/compass/blob/57e7ff75dfaf2ba2268fd651f41eaef16d9ba858/packages/compass-logging/src/index.ts#L79

I think this is currently the only place where it’s potentially accessed outside of the renderer process.